packit/cli/init.py

```diff
@@ -105,3 +144,84 @@ def generate_config(
         logger.debug(f"Packit config file '{config_file}' changed.")
 
     return output_config
+
+
+def generate_precommit_config(config_file: Path):
+    """
+    Parse .pre-commit-config.yaml and generate packit-specific configuration
+    to it if not already present
+    """
+    yaml = ruamel.yaml.YAML()
+    data = yaml.load(config_file)
+
+    if not is_file_empty(config_file):
+        if is_packit_precommit_config_present(data):
+            return
+
+        data = prepare_precommit_config(data)
+
+    else:
+        # add configuration to empty file
+        data = {"repos": [PRECOMMIT_CONFIG]}
+
+    with open(config_file, "w") as f:
+        yaml.dump(data, f)
+
+    logger.debug(f"Pre-commit config file '{config_file}' changed.")
+
+
+def is_packit_precommit_config_present(data):
+    """
+    Check whether packit-specific configuration is already
+    present in .pre-commit-config.yaml
+    """
+    if not isinstance(data, dict):
+        return False
+
+    if "repos" not in data:
+        return False
+
+    if not isinstance(data["repos"], list):
+        return False
+
+    repos = data["repos"]
+    return any(
+        repo.get("repo") == PRECOMMIT_CONFIG.get("repo")
+        and repo.get("hooks") == PRECOMMIT_CONFIG.get("hooks")
```

Comment:
If we were to add more hooks, this could become a problem; could you at least left a comment above it?


```suggestion
        # [NOTE] If we introduce more hooks, make sure to check for subset rather than equality.
        # Or drop the check entirely.
        and repo.get("hooks") == PRECOMMIT_CONFIG.get("hooks")
```